Use raw strings for regexes in otelcol docs (#5636)
grafana/agent: docs/sources/flow/reference/components/otelcol.processor.transform.md

  * `sum([1, 2, 3, 4]) + (10 / 1) - 1`

{{% admonition type="note" %}}
There are two ways of inputting strings in River configuration files:
* Using quotation marks ([normal River strings][river-strings]). Characters such as `\` and

>   `"` will need to be escaped by preceding them with a `\` character.

Reviewer: ```suggestion
  `"` must be escaped by preceding them with a `\` character.
```

> * Using backticks ([raw River strings][river-raw-strings]). No characters will need to be escaped.

Reviewer: ```suggestion
* Using backticks ([raw River strings][river-raw-strings]). No characters must be escaped.
```

>   However, it will not be possible to have backticks inside the string.

Reviewer: ```suggestion
  However, it's not possible to have backticks inside the string.
```
